### PR TITLE
Readonly and inputgroup styles

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/numberfield/src/NumberField.tsx [hunk 1]
```diff
@@ -63,13 +65,16 @@ function NumberField(props: SpectrumNumberFieldProps, ref: FocusableRef<HTMLElem
     {
       'spectrum-InputGroup--quiet': isQuiet,
       'is-invalid': props.validationState === 'invalid',
```
Comment (LFDanLu) [inline packages/@react-spectrum/numberfield/src/NumberField.tsx:67]: Think this needs to be `'spectrum-InputGroup--invalid': validationState === 'invalid'`, noticed the following when on mobile and focus is within an invalid numberfield:
image

Comment (LFDanLu) [reply packages/@react-spectrum/numberfield/src/NumberField.tsx:67]: same for non mobile: 
image

File: packages/@react-spectrum/numberfield/src/NumberField.tsx [hunk 2]
```diff
@@ -146,9 +150,31 @@ const NumberFieldInput = React.forwardRef(function NumberFieldInput(props: Numbe
...
+            )
+          }
           isQuiet={isQuiet}
-          inputClassName={classNames(stepperStyle, 'spectrum-Stepper-input')}
+          inputClassName={
+            classNames(
+              stepperStyle,
+              'spectrum-Stepper-input',
+              classNames(inputgroupStyles,
+                'spectrum-InputGroup-input'
```
Comment (LFDanLu) [inline packages/@react-spectrum/numberfield/src/NumberField.tsx:168]: do we want the right border of the input to be 0 when readonly (this comes from `spectrum-InputGroup-input`)? Perhaps the button's border color should be still be visible?

image

**Review by LFDanLu — CHANGES_REQUESTED** [review]

File: packages/@react-spectrum/numberfield/src/NumberField.tsx (hunk 2 above)
Comment (snowystinger) [reply packages/@react-spectrum/numberfield/src/NumberField.tsx:168]: I opted to follow combobox' lead on this one, might be worth talking to them about

**Review by devongovett — APPROVED** [review]
Looks like it works. Let's run chromatic though to make sure we didn't break anything.

**Comment by snowystinger** [comment]
We don't have numberfield in chromatic yet :-/ I'm working on those stories now

**Review by LFDanLu — APPROVED** [review]
LGTM, happy for the followup w/ Spectrum to handled outside this pull

**Comment by devongovett** [comment]
Ok, I was more thinking of combobox but doesn't look like you touched input group css.

**Pull request merged**